src/Eval.hs

```diff
@@ -813,18 +813,21 @@ printC xobj =
 
 buildMainFunction :: XObj -> XObj
 buildMainFunction xobj =
-  XObj (Lst [ XObj (Defn Nothing) (Just dummyInfo) Nothing
-            , XObj (Sym (SymPath [] "main") Symbol) (Just dummyInfo) Nothing
-            , XObj (Arr []) (Just dummyInfo) Nothing
-            , case ty xobj of
-                Just UnitTy -> xobj
-                Just (RefTy _ _) -> XObj (Lst [XObj (Sym (SymPath [] "println*") Symbol) (Just dummyInfo) Nothing, xobj])
-                                       (Just dummyInfo) (Just UnitTy)
-                Just _ -> XObj (Lst [XObj (Sym (SymPath [] "println*") Symbol) (Just dummyInfo) Nothing,
-                                     XObj (Lst [XObj Ref (Just dummyInfo) Nothing, xobj])
-                                           (Just dummyInfo) (Just UnitTy)])
-                               (Just dummyInfo) (Just UnitTy)
-            ]) (Just dummyInfo) (Just (FuncTy [] UnitTy StaticLifetimeTy))
+  XObj (Lst [ XObj (Defn Nothing) di Nothing
```

Comment:
What is this fix about?

Reply:
If you try at the REPL something like `(IO.println "42")` with `tcc` it will fail. If you don't set explicitly the return value to 0, the return value of that expressions will just be whatever the `rax` register contains (in this case, IIRC it was the length of the printed string). This wraps the expression in `(do xxx 0)`.

Reply:
Even though it shouldn't have any adverse effects for other compilers/systems, I'm a bit wary of baking in a solution for the peculiarities of one particular system right into the compiler. Even though it doesn't have a behavioral cost, it does increase the complexity of the compiler code slightly and since it's such a specific case we're addressing, I could easily envision our knowledge of why this was added being lost in the far future.

Would it be possible to solve this in Carp instead, using, e.g. a library?

I think it's OK to do this if there's really no other solution. I'd add a brief explanatory comment just so we don't forget why this is here and accidentally delete it in some code cleanup in the future.